File: 1 - metodosBusca/pacoteBusca_JomiHubner/src/busca/BuscaLargura.java
package busca;

import java.util.PriorityQueue;
import java.util.Queue;

/**
 *
 * Busca a solução por busca em largura.
 *
 *  @author Jomi Fred Hübner
 */
public class BuscaLargura extends Busca {
    
    /** busca sem mostrar status */
    public BuscaLargura() {
    }
    
    /** busca mostrando status */
    public BuscaLargura(MostraStatusConsole ms) {
        super(ms);
    }

    public Nodo busca(Estado inicial) {
        status.inicia();
        initFechados();
       
        Queue<Nodo> abertos = new PriorityQueue<Nodo>();
        
        abertos.add(new Nodo(inicial, null));
        
        while (!parar && abertos.size() > 0) {
            
            //System.out.print("abertos "+abertos);
            Nodo n = abertos.remove();
            //System.out.println("pegando "+n);
            status.explorando(n, abertos.size());
            if (n.estado.ehMeta()) {
                status.termina(true);
                return n;
            }
                        
            abertos.addAll( sucessores(n) );
        }
        status.termina(false);
        return null;
    }
    
    public String toString() {
    	return "BL - Busca em Largura";
    }
}

File: 1 - metodosBusca/python/app/core/busca/busca_largura.py
from collections import deque
from .busca import Busca
from .nodo import Nodo

class BuscaLargura(Busca):
    def __init__(self, mostra_status_console=None):
        super().__init__(mostra_status_console)

    def busca(self, inicial):
        self.status.inicia()
        self.init_fechados()
       
        abertos = deque([Nodo(inicial, None)])
        
        while not self.parar and abertos:
            n = abertos.popleft()
            self.status.explorando(n, len(abertos))
            if n.estado.eh_meta():
                self.status.termina(True)
                return n
                        
            abertos.extend(self.sucessores(n))
        
        self.status.termina(False)
        return None
    
    def __str__(self):
        return "BL - Busca em Largura"